### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by t-bello7**
I implemented the following functionalities :
- Users can register 
- User logs in with a combination of email and password.
- Hashed password should be stored in the database.
- Ask for confirmation by email.
- User can reset password

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,7 +1,15 @@
...
-  before_action :set_current_user
+  before_action :configure_permitted_parameters, if: :devise_controller?
 
-  def set_current_user
-    Current.user = User.first
+  # private
+
+  # def after_sign_out_path_for(resource_or_scope)
+  #     root_path
+  # end
```
Comment (Davidosky007) [inline app/controllers/application_controller.rb]: * Kindly remove all commented code as it makes your code base bulky and unprofessional 🙏

File: app/models/user.rb [hunk 2]
```diff
@@ -1,4 +1,8 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (Davidosky007) [inline app/models/user.rb]: * Kindly ensure your devise setup Asks for confirmation of email during signup as stated in [the requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_devise.md#:~:text=Ask%20for%20confirmation%20of%20email.) 
* Also when i try to reset my password i get this error `Missing host to link to! Please provide the :host parameter, set default_url_options[:host], or set :only_path to true`, Kindly fix this error [Following this devise setup instructions helped fix this error](https://github.com/heartcombo/devise#:~:text=config.action_mailer.default_url_options%20%3D%20%7B%20host%3A%20%27localhost%27%2C%20port%3A%203000%20%7D)
image

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @t-bello7 @GordonOtieno 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
## Highlights
* No linter errors 👍 
* Proper use of Gitflow 👍 
* Well written PR description 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 2 above)
Comment (prolajumokeoni) [inline app/models/user.rb]: - [ ]  Based on the requirements of this project, this app should ask for email confirmation please enable  `:confirmable`  in your user.rb
- Just as the previous reviewer requested, please  ensure  your  application asks for confirmation of email, currently I can signin without confirming my email

**Review by prolajumokeoni — CHANGES_REQUESTED** [review]
Hi @t-bello7 and @GordonOtieno 
Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## Highlights
✔️ No linter errors
✔️ Git flow was correctly used  

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- Kindly ensure that as partners you always have balanced commits in a pair programming project 👍 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by GzimAsani — COMMENTED** [review]
# STATUS: APPROVED :100:         
## There is nothing else to say than it's time to merge... :partying_face: 
Cheers and Happy coding! :clap:  :computer:  :wine_glass: 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


## Optional suggestions


_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A
### Please be reminded to not open a new pull request after you get a review, only use the same one :thought_balloon: 


 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

### PS: I think I forgot to hit submit the review when I approved your project. I am glad you figured it out :joy:

**Pull request merged**